ui/components/ui/token-input/token-input.component.js

```diff
@@ -80,8 +80,12 @@ export default class TokenInput extends PureComponent {
 
     let newDecimalValue = decimalValue;
 
-    if (decimals && applyDecimals) {
-      newDecimalValue = parseFloat(decimalValue).toFixed(decimals);
+    if (newDecimalValue.split('.').length > 1) {
+      const decimalsPlaces = newDecimalValue.split('.')[1].length;
+
+      if (decimals && applyDecimals && decimalsPlaces > decimals) {
+        newDecimalValue = parseFloat(decimalValue).toFixed(decimals);
+      }
```

Comment:
@tmashuang : could you replicate the issue and test fix with this PR. `hexValue` does not change input value it changes only value in transaction.

Reply:
Does this test case clarify? https://github.com/MetaMask/metamask-extension/pull/12773/commits/cfe86346e419f8ec89c1c1f13b1cc798b52ed5fa

Reply:
Can we assign `newDecimalValue.split('.')` to a variable so we aren't doing it twice?

Reply:
Done in 5bd45a4, lmk if you figure out a better var name 😅 e